src/api/rate_limiter.py

```diff
@@ -0,0 +1,115 @@
+import asyncio
+import logging
+import uuid
+from datetime import datetime
+from os import getenv
+import requests
+
+DEFAULT_MAX_RETRY = 3
+
+# pylint: disable=too-few-public-methods
+class RequestEntry() :
+    """
+    Class for request object to put in the queue for the rate limiter
+    """
+    def __init__(self,url,cookies,key,method) -> None:
+        self.url = url
+        self.cookies = cookies
+        self.key = key
+        self.method = method
+
+class RateLimiter() :
+    """
+    Class that takes care of sending the requests at a maximum rate (25/sec) 
+        and giving back the result to the calling functions when it is received (async probably)
+    """
+
+    def __init__(self) :
+        self.requests = {}
+        self.queue = asyncio.Queue()
+        #set a max_retry cap
+        if getenv("MAX_API_RETRY") is not None :
+            self._max_retry = int(getenv("MAX_API_RETRY"))
+        else :
+            self._max_retry = DEFAULT_MAX_RETRY
+
+        asyncio.create_task(self.handle_requests())
+
+
+    async def handle_requests(self) :
+        logging.info("Starting rate_limiter task...")
+
+        #local variable
+        last_time_request = datetime.now()
+        retry = False
+        retry_count = 0
+
+        while True :
+            #take a new request from the queue
+            if not retry :
+                request = await self.queue.get()
+                logging.debug(
+                    "Treating item in queue : %s -> %s + %s ",
+                    request.key,request.url,request.cookies
+                    )
+                retry_count = 0
+            else :
+                #request stays the same
+                logging.debug(
+                    "Retrying %s item in queue : %s -> %s + %s ",
+                    retry_count,request.key,request.url,request.cookies
+                    )
+                retry = False
+
+            # wait 50ms for rate limitation purpose ;)
+            loop_duration = (datetime.now() - last_time_request).total_seconds()
+            if (0.050 - loop_duration) > 0.01 :
```

Comment:
I thought that too but actually it's constraint by the rootme api itself if I'm not mistaken, there is then no point in making it configurable.

Reply:
I'll let @Xlitoni close the thread so he can read it

Reply:
I think we can leave it like that, firstly because I took a pretty generous timeout, and secondly because this is just for style, I don't really see a use-case where we would have to spam the API so much that this limit would actually be reached.